# CHM1 simulation

# the training nanopore data nanopore_03262018.fastq.gz can be downloaded at: https://doi.org/doi:10.7282/t3-zw94-js46.
 
# create training fasta
seqtk seq -a nanopore_03262018.fastq > train.fasta

# train nanosim
NanoSim/src/read_analysis.py -I train.fasta -r b38.fna

# simulate reads from chm1
NanoSim/src/simulator.py linear -r chm1.fna -c training -o nanosim_50 --seed 11 -n 50000000

# RSVSim
# extract chr20 from reference
samtools faidx ref.fna chr20 > chr20.fasta

# use RSVSim to generate SV events
R
# R code
# library(RSVSim)
# simulateSV(genome="chr20.fasta",ins=181,dels=96,chrs="chr20",sizeDels=c(92,1659,3262,322,112,297,129,567,329,332,313,416,52,95,323,313,164,828,71,347,341,226,316,128,287,61,102,295,65,736,60,718,57,382,1551,760,6065,311,157,1294,5968,5750,198,2601,305,1908,672,3894,688,3297,333,133,321,312,700,69,141,185,1198,2724,2514,394,112,121,436,1190,359,1738,173,10938,1090,314,96,322,151,6006,72,1280,224,51,321,110,193,305,319,547,56,874,892,185,248,505,197,77,872,198),sizeIns=c(175,221,78,114,317,1197,213,152,302,4806,202,273,84,136,68,345,275,296,333,305,301,119,259,144,514,409,131,138,1144,316,5054,326,54,771,325,347,89,477,146,176,342,318,1135,166,1452,245,537,159,219,105,79,108,159,377,826,222,958,159,439,429,876,168,94,92,66,327,624,375,1139,102,350,1004,591,487,96,576,616,96,475,238,784,996,108,485,250,346,168,703,765,537,504,1777,762,174,225,2515,570,99,196,338,341,164,49,201,100,204,266,316,93,199,549,77,684,183,321,50,316,41310,141,120,2631,61,66,51,1169,1380,1135,551,195,295,61,76,186,134,176,288,1110,298,539,585,304,106,402,321,151,76,294,810,106,541,434,385,955,144,198,518,48,199,359,322,71,299,402,556,224,73,67,307,1323,61,481,210,108,338,40,108,328,344,525,71,334), seed=10)

# simulate reads from generated chr20
NanoSim/src/simulator.py linear -r sim.fasta -c training -o nanosim_50 --seed 11 -n 1200000
